pull/permission.go

```diff
@@ -0,0 +1,51 @@
+package pull
+
+import (
+	"fmt"
+	"strings"
+)
+
+type RepositoryPermission uint8
```

Comment:
I'm unsure about this permission type. Using a number means we can easily check if a given permission includes another one, for example, that people with admin also have write. On the other hand, I don't use that ability anywhere in the current code and it means we need extra functions to parse and generate string representation.

But some "parsing" might be needed even if using string values to validate that inputs are valid permissions...

Reply:
I left another comment around this, but we currently have some existing string normalization. I don't feel strongly about internally representing these as numbers or strings, but I think we should be consistent about it.

Reply:
I decided to stick with integers and ended up using the permission level comparison.